packages/angular/src/migrations/update-12-3-0/convert-webpack-browser-build-target-to-delegate-build.ts

```diff
@@ -0,0 +1,176 @@
+import {
+  formatFiles,
+  getProjects,
+  NxJsonProjectConfiguration,
+  parseTargetString,
+  ProjectConfiguration,
+  Target,
+  TargetConfiguration,
+  targetToTargetString,
+  Tree,
+  updateProjectConfiguration,
+} from '@nrwl/devkit';
+
+export default async function convertWebpackBrowserBuildTargetToDelegateBuild(
+  host: Tree
+) {
+  const projects = getProjects(host);
+
+  for (const [projectName, project] of projects) {
+    const webpackBrowserTargets = Object.values(project.targets).filter(
+      (target) => target.executor === '@nrwl/angular:webpack-browser'
+    );
+    for (const target of webpackBrowserTargets) {
+      const configurationOptions = getTargetConfigurationOptions(target);
+      const buildTargetName = getBuildTargetNameFromOptions(
+        target.options,
+        configurationOptions
+      );
+      if (buildTargetName) {
+        target.executor = '@nrwl/angular:delegate-build';
+        updateTargetsOptions(project, target, buildTargetName);
+        updateTargetsConfigurations(
```

Comment:
I don't think we need to do anything, except perhaps leave a comment, but at this point we'll be entering a triple-nested `for` loop.

I don't see it causing too many perf issues as each set for each nested loop is less than the original set, but if we do encounter perf issues, at least we'll know to look here.

Reply:
Yep, I noticed that, but realistically, it shouldn't be a problem, there shouldn't be many configurations in a target, nor many projects using this executor. The executor is only used in Angular apps and even in large workspaces, the majority of the projects would be libraries and a handful of apps.